web/src/data-providers/ProjectDataProvider.tsx

```diff
@@ -32,9 +32,49 @@ const Context = createContext<ProjectState>({
  *
  * If reloading is required, call the reload function.
  */
-export function ProjectDataProvider({ children }: any): JSX.Element {
+export function ProjectDataProvider ({ children }: any): JSX.Element {
   const { showMessage } = useMessageBanner()
 
+  const loadProjectVersions = (projectName: string): void => {
+    void (async (): Promise<void> => {
+      try {
+        const versions = await ProjectRepository.getVersions(projectName)
+
+        setProjects((state) => {
+          if (state.projectsWithHiddenVersions == null) {
+            return state
+          }
+
+          const project = state.projectsWithHiddenVersions.find(
+            (p) => p.name === projectName
+          )
+
+          if (project == null) {
+            return state
+          }
+
+          project.versions = versions
+
+          return {
+            projects: ProjectRepository.filterHiddenVersions(
+              state.projectsWithHiddenVersions
+            ),
+            projectsWithHiddenVersions: state.projectsWithHiddenVersions,
+            loadingFailed: state.loadingFailed,
+            reload: loadData
+          }
+        })
+      } catch (e) {
+        console.error(e)
+
+        showMessage({
+          text: `Failed to load project ${projectName}`,
+          type: 'error'
+        })
+      }
+    })()
+  }
+
```

Comment:
This adresses the same comments as above, I think in the end it will be slowwer doing a request/project loading
all versions than loading them in the backend. I would argue to remove this change and just return the versions from the backend. 

Having to do api requests per data item is usually not a good api design.